src/app/component/auth/sign-in-new/sign-in-new.component.ts

```diff
@@ -19,20 +21,91 @@ export class SignInNewComponent implements OnInit {
   private googleImage = SignInIcons;
   private hideShowPasswordImage = SignInIcons;
 
+  public userOwnSignIn: UserOwnSignIn;
+  public loadingAnim: boolean;
+  public emailErrorMessageBackEnd: string;
+  public passwordErrorMessageBackEnd: string;
+  public backEndError: string;
+
   constructor(
     private matDialogRef: MatDialogRef<SignInNewComponent>,
     private userOwnSignInService: UserOwnSignInService,
     private router: Router,
     private authService: AuthService,
-    private googleService: GoogleSignInService
+    private googleService: GoogleSignInService,
+    private localStorageService: LocalStorageService,
   ) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.userOwnSignIn = new UserOwnSignIn();
+    this.loadingAnim = false;
+    this.emailErrorMessageBackEnd = null;
+    this.passwordErrorMessageBackEnd = null;
+    this.backEndError = null;
+    this.localStorageService.userIdBehaviourSubject.subscribe(userId => {
+      if (userId) {
+        this.matDialogRef.close();
+      }
+    });
+  }
+
+  public signIn(userOwnSignIn: UserOwnSignIn) {
+    this.loadingAnim = true;
+    this.userOwnSignInService.signIn(userOwnSignIn).subscribe(
+      (data: UserSuccessSignIn) => {
+        this.loadingAnim = false;
+        this.userOwnSignInService.saveUserToLocalStorage(data);
+        this.localStorageService.setFirstName(data.firstName);
+        this.localStorageService.setFirstSignIn();
+        this.router.navigate(['/welcome'])
+          .then(success => console.log('redirect has succeeded ' + success))
+          .catch(fail => console.log('redirect has failed ' + fail));
+      },
+      (errors: HttpErrorResponse) => {
+        try {
+          errors.error.forEach(error => {
+            if (error.name === 'email') {
+              this.emailErrorMessageBackEnd = error.message;
+            } else if (error.name === 'password') {
+              this.passwordErrorMessageBackEnd = error.message;
+            }
+          });
+        } catch (e) {
+          this.backEndError = errors.error.message;
+        }
+        this.loadingAnim = false;
+      }
+    );
+  }
 
-  private closeSignInWindow(): void {
+  private CloseSignInWindow(): void {
     this.matDialogRef.close();
   }
 
+  signInWithGoogle() {
```

Comment:
Please add access modifiers and specify return type

Reply:
Fixed